scripts/prow-smoke-test.sh

```diff
@@ -63,7 +63,7 @@ echo "CONTAINER_ENGINE=$CONTAINER_ENGINE 🐳"
 echo "CONTAINER_IMAGE=$CONTAINER_IMAGE"
 
 # Get the default container $WORKDIR
-CONTAINER_WORKDIR=$($CONTAINER_ENGINE run --rm $CONTAINER_IMAGE /bin/bash -c 'echo $PWD')
+CONTAINER_WORKDIR=$($CONTAINER_ENGINE run --privileged --rm $CONTAINER_IMAGE /bin/bash -c 'echo $PWD')
```

Comment:
**issue:** I don't think we can run privileged containers in Prow because it allows a root shell on the host. PS: https://docs.docker.com/reference/cli/docker/container/run/#privileged 

**suggestion:** 
- I think we can try running it with the `--cap-add` option, as suggested in the warning. So maybe instead of `--privileged` we can use `--cap-add=DAC_OVERRIDE` based on https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities
- We should also update the command only if it's running on Mac arm64 and not for other architectures where it is working fine.

Reply:
I get `sh: scripts/check-asciidoctor-build.sh: Permission denied` when I run with `--cap-add=DAC_OVERRIDE` on Linux x86/podman.

I took your original suggestion and applied that in the latest. PTAL.